sub_peers.js

```diff
@@ -62,6 +62,7 @@ TChannelSubPeers.prototype.add = function add(hostPort, options) {
     }
 
     peer = topChannel.peers.add(hostPort, options);
+    peer.setPreferConnectionDirection(self.preferConnectionDirection);
```

Comment:
We are mutating the preference here. Which means any co-tenant peers across sub channels better have the same preference on all sub channels.

This is last write wins semantics. I would prefer a more immutable first write is final approach.